### PR TITLE
Ansbranch

### PR DESCRIPTION
**Pull request by idhant96**
Answeres were added to answer.txt, Please review the answers and merge the branch.

### CODE REVIEW

File: answer.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+• What is a gitignore file and what should it include for a Java project?
+
+A. It is a hidden file which stores the file names or file formats which git should ignore for checking changes during staging/ modifying the file contents.
+For a typical java project it should include .class files.
+
+• What does fast-forward in Git mean?
+A. When we merge a branch which has more commits than another branch, upon merging the base branch gets updated in a 'fast-forward' method.
+• Why should you have different branches?
+A. To maintain different versions of code.
```
Comment (Gayathri95) [inline answer.txt:12]: Optional comment. It would be better if you could elaborate more on this.

File: answer.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+• What is a gitignore file and what should it include for a Java project?
+
+A. It is a hidden file which stores the file names or file formats which git should ignore for checking changes during staging/ modifying the file contents.
+For a typical java project it should include .class files.
+
+• What does fast-forward in Git mean?
+A. When we merge a branch which has more commits than another branch, upon merging the base branch gets updated in a 'fast-forward' method.
```
Comment (Gayathri95) [inline answer.txt:7]: For this too a better explanation might be helpful.

**Comment by Gayathri95** [comment]
Kindly address comments and merge. Rest all looks good.

File: answer.txt (hunk 1 above)
Comment (idhant96) [reply answer.txt:12]: That should be enough.

File: answer.txt (hunk 2 above)
Comment (idhant96) [reply answer.txt:7]: Okay

**Pull request merged**